jgrapht-core/src/main/java/org/jgrapht/util/FibonacciHeap.java

```diff
@@ -254,6 +254,13 @@ public void insert(FibonacciHeapNode<T> node, double key)
 
             // decrement size of heap
             nNodes--;
+
+            // clear z
+            z.left = z;
+            z.right = z;
+            z.degree = 0;
+            z.child = null;
+            z.mark = false;
```

Comment:
Yep, parent is already null, because `deleteMin()` always removes a node which is the root of a heap-ordered tree.